modules/apps/sharing/sharing-web/src/main/resources/META-INF/resources/shared_assets/js/SharedAssetsManagementToolbarPropsTransformer.js

```diff
@@ -0,0 +1,64 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+import {addParams, navigate, openSelectionModal} from 'frontend-js-web';
+
+export default function propsTransformer({
+	additionalProps: {selectAssetTypeURL, viewAssetTypeURL},
+	filterDropdownItems,
+	portletNamespace,
+	...otherProps
+}) {
+	Liferay.on('sharing:changed', () =>
+		Liferay.Portlet.refresh('#p_p_id' + portletNamespace)
+	);
+
+	return {
+		...otherProps,
+		filterDropdownItems: filterDropdownItems.map((item) => {
+			return {
+				...item,
+				items: item.items.map((childItem) => {
+					return {
+						...childItem,
+						onClick() {
```

Comment:
This code would work, but it's not consistent with component API.

We should create the same pattern here as for [actionDropdownItems](https://github.com/liferay/liferay-portal/blob/master/modules/apps/frontend-taglib/frontend-taglib-clay/src/main/resources/META-INF/resources/management_toolbar/ActionControls.js#L73-L89). With this, we would:
- declare `onClick` internaly 
- have a new callback, `onFilterDropdownItemClick`
- override that callback here in `propsTransformer`.

Note that we don't need to have `clone` or `dataAttributes` complications, as in the linked block. Just a call to the new callback.

Reply:
Hey @markocikos,

I've got the exact same use case for `document-library-web` so I'll work on that and hopefully submit it in short while.